apis/v1alpha1/opentelemetrycollector_types.go

```diff
@@ -291,6 +291,16 @@ type OpenTelemetryCollectorSpec struct {
 	DeploymentUpdateStrategy appsv1.DeploymentStrategy `json:"deploymentUpdateStrategy,omitempty"`
 }
 
+// PortsSpec defines the OpenTelemetryCollector's container/service ports additional specifications.
+type PortsSpec struct {
```

Comment:
This ends up being a breaking change, so maybe I was wrong when I said it should live in the alpha api. @open-telemetry/operator-maintainers can you confirm?

Reply:
Is it? Looks like we're just adding fields to the CRDs. I'm personally a bit worried that we're reinventing the wheel with this PortSpec definition, and the community has converged on some standard for this "ContainerPort and ServicePort in one" concept.

Reply:
I put my comment in the wrong spot.

Changing from `Ports []v1.ServicePort json:"ports,omitempty"` to `Ports []PortsSpec json:"ports,omitempty"` is the breaking change I was referring to.

Reply:
hi @swiatekm-sumo 

While I aggree that mixing values for different reasons can be messy, the current `ports` definition already present in the spec is used both to populate the Service object and the ContainerPorts object.

We could do a dedicated new field or something, but then we would have multiples places for settings being used in also multiples places.

Putting in the same `port` context seemed the cleanest place.

Reply:
@TylerHelmuth Can you elaborate on how this is a breaking change? the interface/spec with the client remains the same, minus the new field `hostPort`.

Reply:
@bmiguel-teixeira I agree with your approach here. My point was that this sounds like a problem other operators also had to solve, and I'm wondering if there's some de-facto standard way of doing it.

@TylerHelmuth `PortsSpec` embeds `ServicePort`, so this doesn't look like a breaking change from the perspective of a CRD user. Can you describe a specific situation where this would break someone?

Reply:
I misinterpreted what `ServicePort` represented, I didn't realize it was a full struct.  I agree this is being done in a non-breaking way.

Reply:
yeah I looked around and didn't see anything from the obvious sources...